lib/rgeo/geographic/spherical_feature_methods.rb

```diff
@@ -130,5 +130,36 @@ def length
         inject(0.0) { |sum, geom| sum + geom.length }
       end
     end
+
+    module SphericalPolygonMethods # :nodoc:
+      def centroid
+        return super unless num_interior_rings == 0
+
+        points = coordinates.first
+
+        centroid_lat = 0.0
+        centroid_lng = 0.0
+        signed_area = 0.0
+
+        # Iterate over each element in the list but the last item as it's
+        # calculated by the i+1 logic
+        points[0...-1].each_index do |i|
```

Comment:
We could use `#each_cons` to iterate, and use `#exterior_ring` to get point objects instead of coordinates. Something like this should work:

```ruby
exterior_ring.points.each_cons(2) do |p0, p1|
  area = (p0.x * p1.y) - (p1.x & p2.y)
  # ...  
```

Reply:
Yeah, that is much nicer.